assembly/src/ast/error_msg.rs

```diff
@@ -0,0 +1,32 @@
+use alloc::string::{String, ToString};
+
+use vm_core::crypto::hash::{Blake3_256, Blake3Digest, Digest};
+
+use super::Immediate;
+use crate::Felt;
+
+/// The type of error messages used in MASM assertions.
+pub type ErrorMsg = Immediate<String>;
```

Comment:
I'd make this `Arc<str>` or `Ident`, rather than `Immediate<String>`, unless we're intentionally trying to represent that an `ErrorMsg` can be a constant name as well as literal string.